it/uk/gov/hmrc/fhregistrationfrontend/controllers/ContactPersonControllerISpec.scala

```diff
@@ -0,0 +1,258 @@
+      "Is the contact address in the UK? Yes" +
+      "All fields entered" should {
+      "return 200 with no contact person's details message" in {
+        given
+          .commonPrecondition
+
+        WsTestClient.withClient { client =>
+          val result = client.url(s"$baseUrl/contactPerson")
+            .addCookies(DefaultWSCookie("mdtp", authAndSessionCookie))
+            .withHttpHeaders(xSessionId,
+              "Csrf-Token" -> "nocheck")
+            .post(Map(
+              "firstName"                                     -> Seq("John"),
+              "lastName"                                      -> Seq("Smith"),
+              "jobTitle"                                      -> Seq("Astronaut"),
+              "telephone"                                     -> Seq("0123456789"),
+              "usingSameContactAddress"                       -> Seq("false"),
+              "isUkAddress"                                   -> Seq("true"),
+              "otherUkContactAddress_contactAddress.Line1"    -> Seq("Flat 1"),
+              "otherUkContactAddress_contactAddress.Line2"    -> Seq("5 High Street"),
+              "otherUkContactAddress_contactAddress.Line3"    -> Seq("Worthing"),
+              "otherUkContactAddress_contactAddress.Line4"    -> Seq("West Sussex"),
+              "otherUkContactAddress_contactAddress.postcode" -> Seq("AB1 2YZ")
+            ))
+
+          whenReady(result) { res =>
+            res.status mustBe 200
+            res.body mustBe "Form submitted, with result: ContactPerson(John,Smith,Astronaut,0123456789,None,false,Some(true),Some(Address(Flat 1,Some(5 High Street),Some(Worthing),Some(West Sussex),AB1 2YZ,None,None)),None)"
+          }
+        }
+      }
+    }
+
+    "no data entered" should {
+      "return 400" in {
+        given
+          .commonPrecondition
+
+        WsTestClient.withClient { client =>
+          val result = client.url(s"$baseUrl/contactPerson")
+            .addCookies(DefaultWSCookie("mdtp", authAndSessionCookie))
+            .withHttpHeaders(xSessionId,
+              "Csrf-Token" -> "nocheck")
+            .post(Map(
+              "firstName"               -> Seq.empty,
+              "lastName"                -> Seq.empty,
+              "jobTitle"                -> Seq.empty,
+              "telephone"               -> Seq.empty,
+              "usingSameContactAddress" -> Seq.empty
+            ))
+
+          whenReady(result) { res =>
+            res.status mustBe 400
+            val page = Jsoup.parse(res.body)
+            page.getElementsByClass("govuk-error-summary").text() must include("There is a problem Enter your contact person's first name " +
+              "Enter your contact person's last name " +
+              "Enter your contact person's job title " +
+              "Enter your contact person's telephone number " +
+              "Select whether you want to use the contact address")
+          }
+        }
+      }
+    }
+
```

Comment:
New line character can't be added here as it's an assertion that the error equals that text and there is no line breaks in the error. It would cause the test to fail.